#Análisis de Accidentes Laborales

##Trabajo Final
Diploma Data Analyst

Alumno: Nadir De Los Angeles Díaz Meneses

Fuente: https://www.kaggle.com/datasets/jpmiller/work-accidents-in-china

Objetivo:
Analizar los accidentes laborales e identificar patrones y tendencias.

In [1]:
import pandas as pd
df=pd.read_csv("/content/accidents.csv")
df.head()

,Title,Date,Description,Province,City,Industry,Sub Industry,Company Name,Company Ownership,Type,Number of Deaths,Number of Injuries,Number of Punished,Financial Penalty
0,"A building caught fire in Dalian, Liaoning",2021-08-27,"A building caught fire in Dalian, Liaoning",Liaoning,Dalian,Construction,NaN,NaN,Unknown,Fire,0,0,0,0
1,A construction site caught fire in Beijing,2021-08-26,A construction site caught fire in Beijing,Beijing,NaN,Construction,NaN,NaN,Unknown,Fire,0,0,0,0
2,Four injured after collapse at a highway site ...,2021-08-23,Four injured after collapse at a highway site ...,Yunnan,Honghe,Construction,NaN,NaN,Unknown,Falling/Structural failure,0,1-9,0,0
3,"One dead in accident at a coal mine in Baotou,...",2021-08-21,"One dead in accident at a coal mine in Baotou,...",Neimenggu,Baotou,Coal mining,NaN,NaN,Private Enterprise,Structural failure,1-9,0,0,0
4,A factory of Taiyuan Steel caught fire after t...,2021-08-21,A factory of Taiyuan Steel caught fire after t...,Shanxi,Taiyuan,Industrial(iron and steel),NaN,NaN,SOE,Fire,0,0,0,0


In [2]:
df.shape

(279, 14)

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 279 entries, 0 to 278
Data columns (total 14 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Title               279 non-null    object 
 1   Date                279 non-null    object 
 2   Description         279 non-null    object 
 3   Province            279 non-null    object 
 4   City                276 non-null    object 
 5   Industry            279 non-null    object 
 6   Sub Industry        1 non-null      object 
 7   Company Name        0 non-null      float64
 8   Company Ownership   278 non-null    object 
 9   Type                276 non-null    object 
 10  Number of Deaths    279 non-null    object 
 11  Number of Injuries  279 non-null    object 
 12  Number of Punished  279 non-null    int64  
 13  Financial Penalty   279 non-null    int64  
dtypes: float64(1), int64(2), object(11)
memory usage: 30.6+ KB


In [4]:
df.isnull().sum()

,0
Title,0
Date,0
Description,0
Province,0
City,3
Industry,0
Sub Industry,278
Company Name,279
Company Ownership,1
Type,3


In [5]:
df.duplicated().sum()

np.int64(1)

Se encontró 1 columna duplicada.

In [6]:
df.describe()

,Company Name,Number of Punished,Financial Penalty
count,0.0,279.0,279.0
mean,NaN,0.0,0.0
std,NaN,0.0,0.0
min,NaN,0.0,0.0
25%,NaN,0.0,0.0
50%,NaN,0.0,0.0
75%,NaN,0.0,0.0
max,NaN,0.0,0.0


In [7]:
df.describe(include="object")

,Title,Date,Description,Province,City,Industry,Sub Industry,Company Ownership,Type,Number of Deaths,Number of Injuries
count,279,279,279,279,276,279,1,278,276,279,279
unique,271,139,271,35,147,8,1,6,36,3,4
top,A construction site caught fire in Beijing,2021-04-09,A construction site caught fire in Beijing,Zhejiang,Tianjing,Construction,Sanitation workers,Private Enterprise,Fire,1-9,0
freq,3,7,3,24,20,100,1,125,46,137,181


El top de accidentes ocurridos en China son relacionados a incendios/fuego, hubo una locación donde fallecieron hasta 9 personas. El rubro donde ocurrieron mas accidentes fue en el sector construcción.

In [19]:
import plotly.express as px
fig = px.bar(
    df,
    x ="Industry",
    title="Accidentes por sector")
fig.show()

El sector con mas accidentes es el de construcción triplicando aproximadamente la cantidad del segundo lugar el cual es minería de carbon. Muy cercano esta, en tercer lugar, la industria de hierro y acero.

In [24]:
import plotly.express as px
fig = px.bar(
    df,
    x ="City",
    title="Accidentes por ciudad")
fig.show()

De todas las ciudades, las que presentan mas cantidad de accidentes de trabajo son las ciudades Tianjing y Bejing.

In [20]:
fig = px.histogram(
    df,
    x="Number of Deaths",
    title="Distribución de personas fallecidas"
)

fig.show()

Hay 135 empresas que en el periodo de tiempo del registro solo han presentado accidentes de trabajo con lesiones. Por otro lado, 137 empresas que presentaron accidentes de trabajo con consecuencia de muertes, entre 1 a 9 muertos en cada evento ocurrido. Hay un tercer escenario, donde 7 empresas tuvieron accidentes de trabajo con impactó en cada evento, entre 10 a 29 trabajadores fallecidos.

In [22]:
fig = px.histogram(
    df,
    x="Number of Injuries",
    title="Distribución de personas lesionadas"
)

fig.show()


In [23]:
fig = px.box(
    df,
    y="Financial Penalty",
    title="Distribución de importe de la multa"
)
fig.show()

Por lo que se aprecia, las empresas con accidentes de trabajo o no han registrado su sanción económica por los accidentes de trabajo ocurridos o no ahn sido impuestas por el ente fisclaizador.

In [27]:
!pip install duckdb


In [28]:
import duckdb

In [29]:
import pandas as pd
df=pd.read_csv("/content/accidents.csv")

In [30]:
df.head()

,Title,Date,Description,Province,City,Industry,Sub Industry,Company Name,Company Ownership,Type,Number of Deaths,Number of Injuries,Number of Punished,Financial Penalty
0,"A building caught fire in Dalian, Liaoning",2021-08-27,"A building caught fire in Dalian, Liaoning",Liaoning,Dalian,Construction,NaN,NaN,Unknown,Fire,0,0,0,0
1,A construction site caught fire in Beijing,2021-08-26,A construction site caught fire in Beijing,Beijing,NaN,Construction,NaN,NaN,Unknown,Fire,0,0,0,0
2,Four injured after collapse at a highway site ...,2021-08-23,Four injured after collapse at a highway site ...,Yunnan,Honghe,Construction,NaN,NaN,Unknown,Falling/Structural failure,0,1-9,0,0
3,"One dead in accident at a coal mine in Baotou,...",2021-08-21,"One dead in accident at a coal mine in Baotou,...",Neimenggu,Baotou,Coal mining,NaN,NaN,Private Enterprise,Structural failure,1-9,0,0,0
4,A factory of Taiyuan Steel caught fire after t...,2021-08-21,A factory of Taiyuan Steel caught fire after t...,Shanxi,Taiyuan,Industrial(iron and steel),NaN,NaN,SOE,Fire,0,0,0,0


In [31]:
duckdb.query("""
SELECT *
FROM df
LIMIT 10
""").to_df()


,Title,Date,Description,Province,City,Industry,Sub Industry,Company Name,Company Ownership,Type,Number of Deaths,Number of Injuries,Number of Punished,Financial Penalty
0,"A building caught fire in Dalian, Liaoning",2021-08-27,"A building caught fire in Dalian, Liaoning",Liaoning,Dalian,Construction,None,NaN,Unknown,Fire,0,0,0,0
1,A construction site caught fire in Beijing,2021-08-26,A construction site caught fire in Beijing,Beijing,None,Construction,None,NaN,Unknown,Fire,0,0,0,0
2,Four injured after collapse at a highway site ...,2021-08-23,Four injured after collapse at a highway site ...,Yunnan,Honghe,Construction,None,NaN,Unknown,Falling/Structural failure,0,1-9,0,0
3,"One dead in accident at a coal mine in Baotou,...",2021-08-21,"One dead in accident at a coal mine in Baotou,...",Neimenggu,Baotou,Coal mining,None,NaN,Private Enterprise,Structural failure,1-9,0,0,0
4,A factory of Taiyuan Steel caught fire after t...,2021-08-21,A factory of Taiyuan Steel caught fire after t...,Shanxi,Taiyuan,Industrial(iron and steel),None,NaN,SOE,Fire,0,0,0,0
5,Four trapped and rescued after accident at a c...,2021-08-19,Four trapped and rescued after accident at a c...,Ningxia,Yinchuan,Coal mining,None,NaN,SOE,Structural failure,0,1-9,0,0
6,A tower crane collapsed at a construction site...,2021-08-19,A tower crane collapsed at a construction site...,Fujian,Xiamen,Construction,None,NaN,Unknown,Structural failure/Mechanical accident,0,0,0,0
7,Two dead and two injured in accident at a rese...,2021-08-18,Two dead and two injured in accident at a rese...,Beijing,Beijing,Industrial(iron and steel),None,NaN,Private Enterprise,Mechanical accident,1-9,1-9,0,0
8,Three poisoned by hydrogen sulfide gas at a fi...,2021-08-17,Three poisoned by hydrogen sulfide gas at a fi...,Zhejiang,Ningbo,Other,None,NaN,Private Enterprise,Gas or Chemical Exposure,0,1-9,0,0
9,An asphalt factory caught fire with five injur...,2021-08-17,An asphalt factory caught fire with five injur...,Guangxi,Nanning,Industrial(iron and steel),None,NaN,Private Enterprise,Fire,0,1-9,0,0


In [36]:
duckdb.query("""
SELECT *
FROM df
WHERE Industry ='Construction'
""").to_df()

,Title,Date,Description,Province,City,Industry,Sub Industry,Company Name,Company Ownership,Type,Number of Deaths,Number of Injuries,Number of Punished,Financial Penalty
0,"A building caught fire in Dalian, Liaoning",2021-08-27,"A building caught fire in Dalian, Liaoning",Liaoning,Dalian,Construction,None,NaN,Unknown,Fire,0,0,0,0
1,A construction site caught fire in Beijing,2021-08-26,A construction site caught fire in Beijing,Beijing,None,Construction,None,NaN,Unknown,Fire,0,0,0,0
2,Four injured after collapse at a highway site ...,2021-08-23,Four injured after collapse at a highway site ...,Yunnan,Honghe,Construction,None,NaN,Unknown,Falling/Structural failure,0,1-9,0,0
3,A tower crane collapsed at a construction site...,2021-08-19,A tower crane collapsed at a construction site...,Fujian,Xiamen,Construction,None,NaN,Unknown,Structural failure/Mechanical accident,0,0,0,0
4,Four died after collapse of a bridge construct...,2021-08-17,Four died after collapse of a bridge construct...,Anhui,Hefei,Construction,None,NaN,SOE,Drowning/Structural failure,1-9,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
95,One dead from falling at a construction site i...,2021-03-09,One dead from falling at a construction site i...,Beijing,Beijing,Construction,None,NaN,Private Enterprise,Falling,1-9,0,0,0
96,Two tower cranes collapse at a construction si...,2021-03-08,Two tower cranes collapse at a construction si...,Jiangsu,Yancheng,Construction,None,NaN,Unknown,Mechanical accident,0,0,0,0
97,A tower crane collpased at a construction site...,2021-03-06,A tower crane collpased at a construction site...,Shanxi,"Yan,an",Construction,None,NaN,Unknown,Falling/Mechanical accident,0,0,0,0
98,Three dead in accident at a construction site ...,2021-03-05,Three dead in accident at a construction site ...,Neimenggu,Baotou,Construction,None,NaN,Private Enterprise,Falling/Mechanical accident,1-9,0,0,0


In [37]:
duckdb.query("""
SELECT Industry,
COUNT(*) AS Total
FROM df
GROUP BY Industry
ORDER BY Total DESC
""").to_df()

,Industry,Total
0,Construction,100
1,Coal mining,35
2,Industrial(iron and steel),33
3,Storage/Logistics,32
4,Repair/maintenance,28
5,Other,27
6,Manufacturing,23
7,Other mining industry,1


In [50]:
duckdb.query("""
SELECT "Company Ownership",
AVG("Financial Penalty") AS Promedio
FROM df
GROUP BY "Company Ownership"
ORDER BY Promedio DESC
""").to_df()

,Company Ownership,Promedio
0,Public Institution,0.0
1,None,0.0
2,Foreign,0.0
3,"HK, Taiwan, Macau",0.0
4,Unknown,0.0
5,Private Enterprise,0.0
6,SOE,0.0


#Conclusiones

1. El dataset contiene 279 registros de accidentes ocurridos entre marzo y agosto 2021.

2. El sector construccion registra la mayor cantidad de accidentes.

3. Los accidentes mas frecuentes estan relacionados con incendio, fallas estructurales, accidentes mecánicos, explisiones y exposición a gases químicos.

4. Se identificaron 144 registros con fallecidos y 98 con lesionados, lo cual evidencia que muchos casos tienen consecuencia severas para la seguridad y salud de las personas.

5. Desde el enfoque de prevención de riesgos, los resultados permiten priorizad controles en trabajos de construcción, minería, mantenimiento, logística y actividades industriales donde existen peligros críticos como caídas, fallas estructurales, punto de calor, equipos mecánicos y sustancias químicas.

6. Este análisis demuestra la importancia de utilizar herramientas de analítica de datos ara identificar patrones de accidentabilidad. En el campo de la prevención de riesgos, el uso de Python, SQL y visualizaciones permite convertir datos históricos en información útil para la toma de decisiones, la priorización de controles y la mejora continua de la gestión de seguridad.